### PR TITLE
feat: Refine mutex tree definition, add methods

### PR DESCRIPTION
**Pull request by lmondada**

### CODE REVIEW

**Comment by lmondada** [comment]
Another fairly easy PR to review. The bulk is coming next...

File: src/new_api_v04/mutex_tree.rs [hunk 1]
```diff
@@ -1,27 +1,41 @@
...
+//!   - tree leaves may be labelled with one or more integers, to be interpreted as
+//!     indices with respect to a list of constraints.
+//!   - an ordering of the children for any node.
+//!
+//! Its interpretation is most easily explained by viewing the tree as a state
+//! automaton, where we say that node `n` is reachable for a given input if
+//!  - `n` is the root, or
+//!  - the parent `p` of `n` is reachable, the constraint on the edge from `p`
+//!    to `n` is satisfied and none of the constraints on the edges between `p`
+//!    and a sibling of `n` that is smaller than `n` are satisfied.
```
Comment (aborgna-q) [inline src/new_api_v04/mutex_tree.rs:19]: A more concise rewording. Feel free to ignore.
```suggestion
//!  - the parent `p` of `n` is reachable and `n` is the smallest sibling for 
//!    which the edge constraint from `p` to the sibling is satisfied.
```

File: src/new_api_v04/mutex_tree.rs [hunk 2]
```diff
@@ -31,52 +45,84 @@
...
-        }
+    pub fn add_children<'a>(
+        &'a mut self,
+        node: usize,
+        predicates: impl IntoIterator<Item = P> + 'a,
+    ) -> impl Iterator<Item = usize> + 'a {
+        predicates.into_iter().map(move |p| self.add_child(node, p))
     }
 
     /// Add a child to a node in the tree.
```
Comment (aborgna-q) [inline src/new_api_v04/mutex_tree.rs:124]:
```suggestion
    /// Add a child to a node in the tree.
    ///
    /// Returns the index of the inserted node.
```

**Review by aborgna-q — APPROVED** [review]
👍

**Pull request merged**